airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2109,3 +2128,265 @@ def execute(self, context: Any):
+            self.log.info('All expectations met')
+        else:
+            self.log.info('One or more expectations were not met.')
+            if self.send_alert_email:
+                self.log.info('Sending alert email...')
+                self.send_alert(data_docs_url)
+            if self.fail_if_expectations_not_met:
+                raise AirflowException('One or more expectations were not met')
+
+    # Generate a unique name for a temporary BQ table
+    def get_temp_table_name(self):
+        now = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+        name_start = "temp_great_expectations_" + now + '_'
+        full_name = name_start + ''.join(random.choices(string.ascii_uppercase +
+                                                        string.digits, k=5))
+        log.info("Generated name for temporary table: " + full_name)
+        return full_name
+
+    def send_alert(self, data_docs_url):
+        if self.include_datadocs_link_in_email:
+            # From an Airflow variable set by the user, get the domain name of the service serving the data docs.
+            domain = Variable.get("great_expectations_datadocs_domain")
+            # Replace the domain returned by GE with the domain set up to serve the data docs
+            parsed = urlsplit(data_docs_url)
+            new_url = parsed._replace(netloc=domain)
+            results = '  See the results <a href=' + new_url.geturl() + '>here</a>.'
+        else:
+            # From the data docs url, pull out just the GCS path and send it to the users in the email.
+            parsed = urlsplit(data_docs_url)
+            results = '  See the following GCS location for results:' + parsed.path
+        email_content = '''
+                <html>
+                  <head>
+                    <meta charset="utf-8">
+                  </head>
+                  <body style="background-color: #fafafa; font-family: Roboto, sans-serif=;">
+                    <div style="width: 600px; margin:0 auto;">
+                        <div style="background-color: white; border-top: 4px solid #22a667; border-left: 1px solid #eee; border-right: 1px solid #eee; border-radius: 6px 6px 0 0; height: 24px;"></div>
+                            <div style="background-color: white; border-left: 1px solid #eee; border-right: 1px solid #eee; padding: 0 24px; overflow: hidden;">
+                              <div style="margin-left: 35px;">
+                                Great Expectations Alert<br>
+                                One or more data expectations were not met in the {0} file. {1}
+                           </div>
+                  </body>
+                </html>
+                '''.format(self.expectations_file_name, results)
+        send_email(self.email_to, 'expectations in ' + self.expectations_file_name + ' not met', email_content,
+                   files=None, cc=None, bcc=None,
+                   mime_subtype='mixed', mime_charset='us_ascii')
+
```

Comment:
How about extracting this string to a top level as a constant? This should improve readability of operator logic